app/controllers/events_controller.rb

```diff
@@ -7,6 +7,6 @@ def index
 
   def show
     @event = Event.find_by(id: params[:id])
-    @items = Item.available.active.where.not(id: session[:cart]).where(event_id: @event.id)
+    @items = @event.items.active.not_in_cart(session[:cart])
   end
```

Comment:
Since the active calls the available through events, it was doing double the work for the same result. Also used the not_in_cart scope that was already setup.